src/kernel/src/fs/mod.rs

```diff
@@ -193,13 +193,7 @@ impl Fs {
                 loop {
                     nd.cnd.nameptr += 1;
 
-                    if nd
-                        .cnd
-                        .pnbuf
-                        .get(nd.cnd.nameptr)
-                        .filter(|&v| *v == b'/')
-                        .is_none()
-                    {
+                    if nd.cnd.pnbuf.get(nd.cnd.nameptr).is_some_and(|&v| v == b'/') {
```

Comment:
I don't think the refactored version has the same behavior.

Reply:
Yeah, you're right.

Reply:
Hopefully fixed